### ISSUE TITLE
False positive method-hidden in overridden methods

### ISSUE THREAD

**Issue by craynic** [PyCQA/pylint#3745]
### Steps to reproduce
code:
```python
class Base:
 def __init__(self, func):
  self.func = func
class Extended(Base):
 def __init__(self):
  pass
 def func(self, arg):
  print(arg)
```
The method `__init__` is overridden so it would never hide the method `func`.
### Current behavior
```
a.py:7:1: E0202: An attribute defined in a line 3 hides this method (method-hidden)
```

### Expected behavior
No error.

### pylint --version output
pylint 2.5.3
astroid 2.4.2
Python 3.8.3 (default, May 17 2020, 18:15:42)
